compatibility/test.sh

```diff
@@ -54,10 +54,50 @@ function stop_postgresql() {
     rm -rf "$POSTGRESQL_ROOT_DIR"
   fi
 }
+echo "Starting PostgreSQL database"
 trap stop_postgresql EXIT
 stop_postgresql # in case it's running from a previous build
 start_postgresql
 
+
+# Set up a shared Oracle instance
+# Note: this is code duplicated from ../ci/build.yml
+function start_oracle() {
+  # Only log in automatically if DOCKER_LOGIN is set (as is in our CI).
+  # When running this script locally, the developer is expected to have docker running and authenticated.
+  if [ -z "${DOCKER_LOGIN:-}" ]; then
+    echo "DOCKER_LOGIN is not set, skipping docker login"
+  else
+    docker login --username "$DOCKER_LOGIN" --password "$DOCKER_PASSWORD"
```

Comment:
You need to add
```
      env:
        DOCKER_LOGIN: $(DOCKER_LOGIN)
        DOCKER_PASSWORD: $(DOCKER_PASSWORD)
```
to the compat job similar to what we have for the `linux_oracle` job.